src/legacy/core_plugins/kibana/public/discover/components/histogram/histogram.tsx

```diff
@@ -0,0 +1,265 @@
+
+    return moment(val).format(xAxisFormat);
+  };
+
+  public renderRectAnnotationTooltip = (details?: string) => (
+    <div style={{ minWidth: 200 }}>
+      <EuiFlexGroup alignItems="center">
+        <EuiFlexItem grow={false}>
+          <EuiIcon type="alert" />
+        </EuiFlexItem>
+        <EuiFlexItem>{details}</EuiFlexItem>
+      </EuiFlexGroup>
+    </div>
+  );
+
+  public renderBarTooltip = (xInterval: number, domainStart: number, domainEnd: number) => (
+    headerData: TooltipValue
+  ): JSX.Element | string => {
+    const headerDataValue = headerData.value;
+    const formattedValue = this.formatXValue(headerDataValue);
+
+    const partialDataText = i18n.translate('kbn.discover.histogram.partialData.bucketTooltipText', {
+      defaultMessage:
+        'Part of this bucket may contain partial data. The selected time range does not fully cover it.',
+    });
+
+    if (headerDataValue < domainStart || headerDataValue + xInterval > domainEnd) {
+      return (
+        <React.Fragment>
+          <EuiFlexGroup alignItems="center" className="dscHistogram__header--partial">
+            <EuiFlexItem grow={false}>
+              <EuiIcon type="iInCircle" />
+            </EuiFlexItem>
+            <EuiFlexItem>{partialDataText}</EuiFlexItem>
+          </EuiFlexGroup>
+          <EuiSpacer size="m" />
+          <p>{formattedValue}</p>
+        </React.Fragment>
+      );
+    }
+
+    return formattedValue;
+  };
+
+  public render() {
+    const uiSettings = chrome.getUiSettingsClient();
+    const timeZone = timezoneProvider(uiSettings)();
+    const { chartData } = this.props;
+
+    if (!chartData || !chartData.series[0]) {
+      return null;
+    }
+
+    const data = chartData.series[0].values;
+
+    /**
+     * Deprecation: [interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval].
+     * see https://github.com/elastic/kibana/issues/27410
+     * TODO: Once the Discover query has been update, we should change the below to use the new field
+     */
+    const xInterval = chartData.ordered.interval;
+
+    const xValues = chartData.xAxisOrderedValues;
+    const lastXValue = xValues[xValues.length - 1];
+
+    const domain = chartData.ordered;
+    const domainStart = domain.min.valueOf();
+    const domainEnd = domain.max.valueOf();
+
+    const domainMin = data[0].x > domainStart ? domainStart : data[0].x;
+    const domainMax = domainEnd - xInterval > lastXValue ? domainEnd - xInterval : lastXValue;
+
+    const xDomain = {
+      min: domainMin,
+      max: domainMax,
+      minInterval: xInterval,
+    };
+
+    // Domain end of 'now' will be milliseconds behind current time, so we extend time by 1 minute and check if
+    // the annotation is within this range; if so, the line annotation uses the domainEnd as its value
+    const now = moment();
+    const isAnnotationAtEdge = domainEnd + 60000 > now && now > domainEnd;
```

Comment:
I think the moment API could be helpful here or at least make this logic more clear. After all, we are depending on moment, might as well use it to the fullest.

Such as [`isAfter`](https://momentjs.com/docs/#/query/is-after/) or [`isBefore`](https://momentjs.com/docs/#/query/is-before/)

Reply:
See https://github.com/elastic/kibana/pull/36517/commits/ce5c42d6c42bd2e4f64791de32167f5350a4dbdb